src/twisted/internet/endpoints.py

```diff
@@ -1466,6 +1473,13 @@ def _parseSSL(
         dhParameters=dhParameters,
         **kw,
     )
+    if cipher:
+        try:
+            cipherBytes = cipher.replace(",", ":").encode("ascii")
+            cf.getContext().set_cipher_list(cipherBytes)
+        except SSLError:
+            raise Exception("Invalid cipher list passed")
```

Comment:
I don't think that raising a generic exception is a good idea.

And the SSLError details are lost.

-------

I have checked the other parts of twisted, and I see that they just let the low-level SSLError bubble to the caller.

No need for special handling of the error.

Just make sure that we have a test for the error case.

-------

I have checked the Twisted coding standard, but I could not find any guidence for how is best to handle the exceptions

https://docs.twisted.org/en/stable/development/coding-standard.html


--------

But I think that converting an SSLError into a generic Exception is not a good idea.

Reply:
[#e9308d](https://github.com/twisted/twisted/pull/12137/commits/6e30f9f83d118633e7be1c489d00073549c1b285)